core/utils/queue.h

```diff
@@ -0,0 +1,24 @@
+#ifndef BESS_UTILS_QUEUE_H
+#define BESS_UTILS_QUEUE_H
+
+namespace bess {
+namespace utils {
+
+// Takes Template argument T that is the type to be enqueued and dequeued.
+template <typename T>
+class Queue {
+ public:
+  virtual ~Queue(){};
```

Comment:
The comments for methods should be here in this .h file and not in child classes that inherit and override these.